ultralytics/data/annotator.py

```diff
@@ -64,3 +71,337 @@ def auto_annotate(
+        """
+        Initializes the AutoAnnotator with either a YOLO or Florence2 model.
+
+        Args:
+            model (str, optional): Model name or path. Defaults to "yolo11n.pt".
+            device (str): Device to use for inference (e.g., 'cpu', 'cuda:0').
+            variant (str): Florence2 model variant if using Florence2 (e.g., "base", "large-ft").
+        """
+        self.YOLO_MODEL = True  # Assume YOLO by default
+
+        if model is None:
+            LOGGER.warning("⚠️ No model provided. Using default: yolo11n.pt")
+            self.model = "yolo11n.pt"
+        else:
+            model_str = str(model).casefold()
+
+            if model_str.endswith(".pt") and model_str not in ("florence2.pt", "florence2"):
+                self.model = YOLO(model)
+            elif model_str in ("florence2.pt", "florence2"):
+                self.model = Florence2(device=device, variant=variant)
+                self.YOLO_MODEL = False
+            else:
+                LOGGER.warning(
+                    f"⚠️ Unsupported model '{model}'. Using default: yolo11n.pt.\n"
+                    f"✅ Supported models:\n"
+                    f"   - YOLO: yolov8n.pt, yolov11n.pt, yolov10.pt, yolo11n.pt, yolo12n.pt\n"
+                    f"   - Florence2: florence2"
+                )
+                self.model = YOLO("yolo11n.pt")
+
+        if self.YOLO_MODEL:
+            self.names = self.model.names
+
+    @staticmethod
+    def convert_to_yolo(x1, y1, x2, y2, w, h):
+        """
+        Converts bounding box coordinates to YOLO format.
+
+        Args:
+            x1 (float): X-coordinate of the top-left corner of the bounding box.
+            y1 (float): Y-coordinate of the top-left corner of the bounding box.
+            x2 (float): X-coordinate of the bottom-right corner of the bounding box.
+            y2 (float): Y-coordinate of the bottom-right corner of the bounding box.
+            w (int): Image width.
+            h (int): Image height.
+
+        Returns:
+            bbox (str): Bounding box in YOLO format (cx, cy, width, height).
+        """
+        cx, cy, bw, bh = xyxy2xywhn(torch.tensor([[x1, y1, x2, y2]]), w=w, h=h, clip=True)[0].tolist()
+        return f"{cx:.6f} {cy:.6f} {bw:.6f} {bh:.6f}"
+
+    @staticmethod
+    def create_classes_txt(output_dir, label_map):
+        """
+        Creates a `classes.txt` file listing all class names.
+
+        Args:
+            output_dir (Path): Directory where to save the file.
+            label_map (dict): Dictionary of {label_name: class_id}.
+        """
+        try:
+            with open(output_dir / "classes.txt", "w", encoding="utf-8") as f:
+                for label, idx in sorted(label_map.items(), key=lambda x: x[1]):
+                    f.write(f"{label}\n")
+            LOGGER.info(f"✅ Saved class names to {output_dir / 'classes.txt'}")
+        except Exception as e:
+            LOGGER.error(f"❌ Failed to save classes.txt: {e}")
```

Comment:
@RizwanMunawar all emojis need to be removed from LOGGER.error() and LOGGER.warning(), as these have been automatically integrated last week.